docs/naming_conventions.md

```diff
@@ -0,0 +1,71 @@
+
+# Naming Conventions in Kids First FHIR Data Model
+
+This document describes the naming conventions for ids and
+the file names for resources in the Kids First FHIR data model.
+
+## Resource Id
+
+1. A resource in the FHIR model should always have the `id` attribute
+populated by model developers. We should never let the server auto populate
+`id` since it is the primary way that machines and humans will remember,
+reference, and lookup a resource.
+
+2. The `id` value should conform to the the regex pattern in the
+[FHIR spec](https://www.hl7.org/fhir/datatypes.html#id).
```

Comment:
Why don't we spell out the allowed set of characters like `[A-Za-z0-9\-\.]{1,64}`?

Reply:
If that changes on the FHIR spec then our markdown is obsolete. Better to just link directly to the source of truth.